cpp/bench/ann/src/cuvs/cuvs_mg_ivf_pq_wrapper.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cuvs_ann_bench_utils.h"
+#include "cuvs_ivf_pq_wrapper.h"
+#include <cuvs/neighbors/mg.hpp>
+#include <raft/core/resource/nccl_clique.hpp>
+
+namespace cuvs::bench {
+using namespace cuvs::neighbors;
+
+template <typename T, typename IdxT>
+class cuvs_mg_ivf_pq : public algo<T>, public algo_gpu {
```

Comment:
Can't we have a generic mg wrapper for all mg algos?

Reply:
I considered doing this, but it looks like most commonalities are not MG-specific, but rather ANN-specific.

Reply:
Let's think about in the next release how to make this less verbose.